### PR TITLE
HPCC-20087 Ensure helper->onCreate called in keyedjoin service

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

**Comment by ghalliday** [comment]
@jakesmith smoketest is currently failing

**Comment by jakesmith** [comment]
@ghalliday - please review

File: thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp [hunk 1]
```diff
@@ -1021,6 +1027,16 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
                     msg.append(static_cast<std::underlying_type<RecordTranslationMode>::type>(RecordTranslationMode::None));
             }
         }
+        void prepAndSend(CMessageBuffer &msg, CThorExpandingRowArray &processing, unsigned selected, unsigned partNo, unsigned copy)
+        {
+            // JCSMORE - don't _need_ filename in general after 1st call, but avoids challenge/response handling if other side has closed, and relatively small vs msg size
```
Comment (ghalliday) [inline thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp]: Is this comment still valid?

File: thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp [hunk 2]
```diff
@@ -1038,24 +1054,30 @@ class CKeyedJoinSlave : public CSlaveActivity, implements IJoinProcessor
...
-            initRead(msg, selected, partNo, copy);
             unsigned numRows = processing.ordinality();
-            writeRowData(processing, msg);
-
-            if (!comm->send(msg, lookupSlave, kjServiceMpTag, LONGTIMEOUT))
-                throw MakeActivityException(&activity, 0, "CKeyLookupRemoteHandler - comm send failed");
 
-            msg.clear();
+            CMessageBuffer msg;
+            // JCSMORE - don't _need_ filename in general after 1st call, but avoids challenge/response handling if other side has closed, and relatively small vs msg size
```
Comment (ghalliday) [inline thorlcr/activities/keyedjoin/thkeyedjoinslave.cpp]: And this one?

**Comment by ghalliday** [comment]
@jakesmith changes look ok.  See a couple of comments on comments.  Please squash ready for merging.

**Comment by jakesmith** [comment]
@ghalliday - removed redundant comments and squashed.

**Pull request merged**